src/PrestaShopBundle/Service/MultishopContextSwitcher.php

```diff
@@ -23,13 +23,31 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
+declare(strict_types=1);
 
-header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
-header('Last-Modified: '.gmdate('D, d M Y H:i:s').' GMT');
+namespace PrestaShopBundle\Service;
 
-header('Cache-Control: no-store, no-cache, must-revalidate');
-header('Cache-Control: post-check=0, pre-check=0', false);
-header('Pragma: no-cache');
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use Shop;
 
-header('Location: ../../../../../../../../');
-exit;
+class MultishopContextSwitcher
```

Comment:
I put this class in `src/PrestaShopBundle/Service/` because it modifies prestashop business logic through the use of `Shop::setContext()` AND cookie through `context->cookie`. So it's an hybrid, it does belong to both application layer and domain layer. :/